crates/zksync/core/src/cheatcodes.rs

```diff
@@ -183,10 +183,15 @@ pub fn set_mocked_account<'a, DB>(
     address: Address,
     db: &'a mut DB,
     journaled_state: &'a mut JournaledState,
+    caller: Address,
 ) where
     DB: Database,
     <DB as Database>::Error: Debug,
 {
+    if address == caller {
+        tracing::error!("using `mockCall` cheatcode on caller isn't supported in zkVM");
+    }
```

Comment:
Wondering if this check should stay outside since we're not using this value here. Mostly evaluating the implications of an interface where we need to provide the caller, if we don't need it or if we don't have it.

Reply:
Yeah I thought so also, but this is zkVM specific that's why I wanted to put it here